integrations/ollama/src/haystack_integrations/components/generators/ollama/chat/chat_generator.py

```diff
@@ -183,29 +219,29 @@ def __init__(
             A callback function that is called when a new token is received from the stream.
             The callback function accepts StreamingChunk as an argument.
         :param tools:
-            A list of tools or a Toolset for which the model can prepare calls.
-            This parameter can accept either a list of `Tool` objects or a `Toolset` instance.
+            A list of `haystack.tools.Tool` or a `haystack.tools.Toolset`. Duplicate tool names raise a `ValueError`.
             Not all models support tools. For a list of models compatible with tools, see the
             [models page](https://ollama.com/search?c=tools).
         :param response_format:
             The format for structured model outputs. The value can be:
             - None: No specific structure or format is applied to the response. The response is returned as-is.
             - "json": The response is formatted as a JSON object.
-            - JSON Schema: The response is formatted as a JSON object that adheres to the specified JSON Schema.
+            - JSON Schema: The response is formatted as a JSON object that adheres to the specified JSON Schema. (needs Ollama ≥ 0.1.34)
         """
-
         _check_duplicate_tool_names(tools)
 
-        self.timeout = timeout
-        self.generation_kwargs = generation_kwargs or {}
-        self.url = url
         self.model = model
+        self.url = api_base_url or url
+        self.generation_kwargs = generation_kwargs or {}
+        self.timeout = timeout
         self.keep_alive = keep_alive
         self.streaming_callback = streaming_callback
         self.tools = tools
         self.response_format = response_format
+
         self._client = Client(host=self.url, timeout=self.timeout)
 
+    # Serialisation helpers
```

Comment:
I don't think this comment is necessary, could we remove it?

Reply:
done